app/controllers/ems_common.rb

```diff
@@ -437,6 +437,28 @@ def button
         end
         return
       end
+      if params[:pressed] == "ems_cloud_recheck_auth_status"
+        @record = find_by_id_filtered(model, params[:id])
+        authentications = @record.authentication_for_summary
+        authentications.collect do |auth|
+          result, details = @record.authentication_check_queue(auth[:authtype], :save => true)
+          label =
+              case auth[:authtype]
+                when "default" then _("Default")
+                when "metrics" then _("C & U Database")
+                when "amqp"    then _("AMQP")
+                else;               _("<Unknown>")
```

Comment:
We should have the right abstraction here though.  But, when we're duplicating logic like this... we should ask if this is concept that needs an abstraction.